source/common/http/http2/codec_impl.cc

```diff
@@ -118,6 +122,40 @@ void ConnectionImpl::StreamImpl::encodeTrailers(const HeaderMap& trailers) {
     parent_.sendPendingFrames();
   }
 }
+void ConnectionImpl::StreamImpl::readDisable(bool disable) {
+  ENVOY_CONN_LOG(debug, "Stream {} disabled: disable {}, unconsumed_bytes {}", parent_.connection_,
+                 stream_id_, disable, unconsumed_bytes_);
+  if (disable) {
+    ++buffers_overrun_;
+  } else {
+    --buffers_overrun_;
+    if (!buffers_overrun()) {
+      nghttp2_session_consume(parent_.session_, stream_id_, unconsumed_bytes_);
+      unconsumed_bytes_ = 0;
+      parent_.sendPendingFrames();
+    }
+  }
+}
+
+void ConnectionImpl::StreamImpl::pendingRecvBufferHighWatermark() {
```

Comment:
Is it worth heaving temporary asserts for all 4 of these functions like we did in the other PR with a TODO to clean them up after we get more prod experience? I would be in favor of that personally. This is all super scary.